### PR TITLE
Pickup animation will now play on all clients

### PR DESCRIPTION
**Pull request by MaxIsJoe**
### Purpose
title

### Changelog:
CL: pickup animations will now play on all clients.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs [hunk 1]
```diff
@@ -205,6 +208,15 @@ private IEnumerator serverPreformInteractionLogic(HandApply interaction)
...
 
+	[ClientRpc]
+	private IEnumerator RpcPickupAnimation(Transform interactor)
+	{
+		LeanTween.move(gameObject, interactor, pickupAnimSpeed);
+		LeanTween.scale(gameObject, new Vector3(0, 0), pickupAnimSpeed);
+		yield return WaitFor.Seconds(pickupAnimSpeed);
+		LeanTween.scale(gameObject, new Vector3(1, 1), 0.1f);
+	}
+
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs:228]: I dont think that'll start the coroutine on client, maybe switch it to:

<code>

        [ClientRpc]
	private void RpcPickupAnimation(Transform interactor)
	{
		StartCoroutine(RpcPickupAnimation(interaction.Performer.transform));
	}

       private void PickUpAnimation(Transform interactor)
       {
            LeanTween.move(gameObject, interactor, pickupAnimSpeed);
		LeanTween.scale(gameObject, new Vector3(0, 0), pickupAnimSpeed);
		yield return WaitFor.Seconds(pickupAnimSpeed);
		LeanTween.scale(gameObject, new Vector3(1, 1), 0.1f);
       }
</code>

<strike>-Also means you can remove the duplicated animation in serverPreformInteractionLogic and start the server animation with StartCoroutine(RpcPickupAnimation(interaction.Performer.transform)); as well </strike>

 Actually might need that delay so the object isnt hidden before animation has finished.

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs:228]: Also i dont think you'll be able to send Transform in the Rpc, think you'll be able to send gameObject and then do gameObject.Transform on client side.

**Comment by MaxIsJoe** [comment]
Needs some testing now on more than two clients.

File: UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs [hunk 2]
```diff
@@ -167,6 +167,9 @@ private IEnumerator serverPreformInteractionLogic(HandApply interaction)
 
 		LeanTween.scale(gameObject, new Vector3(1, 1), 0.1f);
 
+		//Start the animation on all clients.
+		StartCoroutine(PickupAnim(interaction.Performer.gameObject));
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs]: Might be better to move this to before the server animation as well

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs]: Did you test it?

**Comment by corp-0** [comment]
I don't like how the animation code is repeated for client and server, it could be encapsulated in a single method called by both server and the RPC method.

**Comment by MaxIsJoe** [comment]
> I don't like how the animation code is repeated for client and server, it could be encapsulated in a single method called by both server and the RPC method.

done.

File: UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs [hunk 3]
```diff
@@ -205,6 +208,25 @@ private IEnumerator serverPreformInteractionLogic(HandApply interaction)
...
+	private void PickupAnim(GameObject interactor)
+	{
+		LeanTween.move(gameObject, interactor.transform, pickupAnimSpeed);
+		LeanTween.scale(gameObject, new Vector3(0, 0), pickupAnimSpeed);
+	}
+
+	[ClientRpc]
+	private void RpcPickupAnimation(GameObject interactor)
+	{
+		LeanTween.move(gameObject, interactor.transform, pickupAnimSpeed);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs]: You're still repeating the code, I intended to mean something like this:

```suggestion
		PickUpAnim( interactor);
```

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs]: done

**Pull request merged**